Create UI for expression evaluator

aprajit99 opened the pull request:
- Created the UI for Expression Evaluator.
- Got the code reviewed by teammates and made the suggested changes.
- Will be working on Unit Testing now and later will look into how to integrate it into the Olympus application.

aprajit99 committed:
Created dotnet core project for expression evaluator

aprajit99 committed:
Stiching up the view model viewmodel

aprajit99 committed:
Renaming steps as variables

aprajit99 committed:
Provided the add variable functionality

aprajit99 committed:
Provided the variable input functionality

aprajit99 committed:
Provided the edit variable functionality

aprajit99 committed:
Validate formula and run test functionality added

aprajit99 committed:
Null value inputs supported

aprajit99 committed:
Operator category and description added and extracted from expression evaluator

aprajit99 committed:
Date time support added

aprajit99 committed:
Removed static members by implementing singleton helper class

aprajit99 committed:
Changed clear variable input button style

aprajit99 committed:
Added watermark to indicate null variable input

aprajit99 committed:
Replacing lists with dictionaries

aprajit99 committed:
Formatting code

aprajit99 committed:
Implemented relay command to remove multiple commands

aprajit99 committed:
Combining views for add and edit variable

aprajit99 committed:
Removing third party dependency

aprajit99 committed:
Implementing base view model

aprajit99 committed:
Code formatting

aprajit99 committed:
Created expression evaluator host project

aprajit99 committed:
Targeting multiple frameworks

aprajit99 committed:
Making host project to launch the ui

jrob commented:
Oddities

1. Clicking the variable to edit it puts it in the editor. Not sure how to fix this. Maybe add a edit button in the grid? Let's discuss.

Changes:

1. Add a column for for examples and include in classes. 
  a. `sum` example "sum(a, b); sum(a,1); sum(1, 2, 3)"
  b. `mult` example "1 * 2; a * b; a * 3"
2. Validate should be enabled if there is any text in the current formula window.
3. Run test validate and run test.
4. Current formula and test output should be in a fixed width font.

Minor changes:

1. Rename `ExpressionEvaluatorHost` to `ExpressionEvaluatorDemo`
3. Rename `ExpressionEvaluatorUi` to `ExpressionEvaluatorUI`
5.  There are functions that do not have descriptions. 
6. Descriptions should be grammatically correct. Some don't have periods at the end.
7. Description for `&&` is incorrect.
8. Clicking on a operator should make sure there is a single space around the operator.
9. Functions that require `()` should include them.

aprajit99 committed:
Addressing review comments

aprajit99 committed:
Adding description for ToStringFormat function

aprajit99 commented:
Changes Addressed:

1. Added a variable edit button for every variable. Solves the issue of variable being unnecessarily added to the formula when clicking to edit it.
2. "Validate Formula" and "Run Test" button works independently. Now "Run Test" itself validates the formula and then produces the result, hence no need to validate formula every time the formula changes.
3. Single space is now provided around the operator which is added to the formula.
4. Functions requiring `()`  is  included to the formula.
5. Added descriptions for functions which did not have previously.
6. Renamed the projects as suggested.

Changes yet to be addressed:

1. Need confirmation on whether operator usage examples should be added to the "Description" column or a separate column for "Examples" need to be provided, whichever seems better.
